### ISSUE TITLE
Publish for Scala 2.12

### ISSUE THREAD

**Issue by andyscott** [stanch/reftree#22]
Currently this project is only published for Scala 2.11. I would love to use it for generating some documentation in Scala 2.12.

**Comment by stanch** [comment]
Could you try adding `resolvers += Resolver.bintrayRepo("stanch", "reftree")`? The project is published for both 2.11 and 2.12, but the jCenter repo (which sbt uses by default) somehow only shows one version. See #16 and #17.

**Comment by andyscott** [comment]
I get the following error:
```
sbt:root> compile
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom.sha1
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom.sha1
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error] coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
[error] coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_sjs0.6_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_sjs0.6_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
sbt:root> [error] (core / coursierResolutions) coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
[error] (coreJS / coursierResolutions) coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_sjs0.6_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_sjs0.6_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error] Total time: 2 s, completed Jul 24, 2018 7:03:05 PM
```

**Comment by andyscott** [comment]
I get the following error:
```
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom.sha1
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
Downloading https://dl.bintray.com/stanch/reftree/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
Downloading https://dl.bintray.com/stanch/reftree/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom.sha1
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
Downloading https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom.sha1
Downloading https://dl.bintray.com/stanch/reftree/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
Downloading https://dl.bintray.com/stanch/reftree/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom.sha1
[error] coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
[error]             https://dl.bintray.com/stanch/reftree/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
[error] coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_sjs0.6_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_sjs0.6_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error]             https://dl.bintray.com/stanch/reftree/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error] (core / coursierResolutions) coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
[error]             https://dl.bintray.com/stanch/reftree/org/stanch/reftree_2.12/1.2.0/reftree_2.12-1.2.0.pom
sbt:root> [error] (coreJS / coursierResolutions) coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
[error]     org.stanch:reftree_sjs0.6_2.12:1.2.0:
[error]         not found:
[error]             /Users/andyscott/.ivy2/local/org.stanch/reftree_sjs0.6_2.12/1.2.0/ivys/ivy.xml
[error]             https://repo1.maven.org/maven2/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error]             https://dl.bintray.com/stanch/reftree/org/stanch/reftree_sjs0.6_2.12/1.2.0/reftree_sjs0.6_2.12-1.2.0.pom
[error] Total time: 1 s, completed Jul 24, 2018 7:05:27 PM
```

**Comment by andyscott** [comment]
Seems to work if I add `resolvers += Resolver.bintrayRepo("stanch", "reftree")` and `resolvers += Resolver.bintrayRepo("stanch", "maven")`, though.

Could I convince you to publish this to some more standard repositories? I'd like to consume this from some Bazel builds that pull from a proxied maven repo. The proxy repo isn't configured to pull from non-default Scala maven repositories.

**Comment by stanch** [comment]
Sorry, I actually meant `Resolver.bintrayRepo("stanch", "maven")`, but I wrongly copied it from #17.
I’ll try to publish to Maven Central, but it might take a few days. Just opened a request here: https://issues.sonatype.org/browse/OSSRH-41441.

**Comment by stanch** [comment]
Update: the first step is done, https://github.com/stanch/zipper (which is a dependency) is making its way to Maven Central. Tomorrow I should be able to publish this one as well.

**Comment by andyscott** [comment]
\o/
Awesome, thanks for the update!

**Comment by stanch** [comment]
Done! https://oss.sonatype.org/#nexus-search;quick~stanch. Should take a few hours to appear on search.maven.com. Meanwhile, I still need to update the website, but `npm`/`gitbook` is acting up...

**Comment by rajcspsg** [comment]
@stanch - Could you please publish this for scala 2.13 version as well?

**Comment by stanch** [comment]
Hi @rajcspsg and sorry for the late response. First of all, thanks for your interest in this library. I have not been actively working with Scala or following the developments in the community for the past couple of years, so I am not sure what the effort of migrating to 2.13 would be. For example, I suspect that the standard `RefTree` instances need to be updated to work with 2.13 collections. That said, I am happy to accept pull requests that bump both github.com/stanch/zipper and this project to 2.13, update the dependencies, etc. After that, publishing should be a matter of tagging a new version.

**Comment by rajcspsg** [comment]
@stanch - I've updated scala version to 2.12 https://github.com/stanch/reftree/pull/28 . I've tested and it is working fine. 
For 2.13 I need to updated the dependencies. I will work on it and send the PR soon.

**Comment by danicheg** [comment]
I think this can be safely closed since we not only have support for 2.12 but also support for 2.13 is coming soon

**Comment by stanch** [comment]
Published to Maven Central for 2.12 and 2.13: https://github.com/stanch/reftree/actions/runs/9870959685/job/27258462126. Should be live in some minutes.
